pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -397,6 +397,40 @@ func createClusterAccessSecret(clientset kubernetes.Interface, namespace string,
 	return nil
 }
 
+// UpdateClusterAccessSecret updates the rook-ceph-mon secret
+func UpdateClusterAccessSecret(clientset kubernetes.Interface, namespace string, clusterInfo *cephclient.ClusterInfo, ownerInfo *k8sutil.OwnerInfo) error {
+	var err error
+	// store the secrets for internal usage of the rook pods
+	secrets := map[string][]byte{
+		fsidSecretNameKey: []byte(clusterInfo.FSID),
+		MonSecretNameKey:  []byte(clusterInfo.MonitorSecret),
+		CephUsernameKey:   []byte(clusterInfo.CephCred.Username),
+		CephUserSecretKey: []byte(clusterInfo.CephCred.Secret),
+	}
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       AppName,
+			Namespace:  namespace,
+			Finalizers: []string{DisasterProtectionFinalizerName},
+		},
+		Data: secrets,
+		Type: k8sutil.RookType,
+	}
```

Comment:
This is duplicating code from `createClusterAccessSecret()`. Let's find a way to make sure the cluster access secret generation logic is shared.

Reply:
done.